Question 1 (compulsory):
The quantity of items sold varies per transaction. TTR would like to know what the probability is that a
transaction contains more than 2 items sold. Report and explain your analysis and conclusion.

In [2]:
import numpy as np

In [3]:
import pandas as pd


In [4]:
dataset_path = 'dataset_1668303.csv'

In [5]:
data = pd.read_csv (dataset_path)

In [85]:
more_than_two_items = data[data['transaction_qty'] > 2].shape[0]
print ("more_than_two_items:", more_than_two_items )

more_than_two_items: 1435


In [86]:
total_transactions = data.shape[0]
print ("total_transactions:", total_transactions )

total_transactions: 7356


In [88]:
probability = more_than_two_items / total_transactions * 100

In [89]:
print(f"The probability that a transaction contains more than 2 items sold is: {probability:.4f}")

The probability that a transaction contains more than 2 items sold is: 19.5079


Conclusion:
The analysis revealed that about 19.5% of transactions in the dataset contain more than 2 items sold. This means that if you were to randomly select a transaction from this dataset, there's roughly a 1 in 5 chance that this transaction would include more than 2 items. This information can be useful for understanding purchasing behavior, inventory management, and sales strategies. It highlights a pattern of purchasing behavior that could be critical for business decisions, such as stocking levels, promotions, and marketing strategies aimed at increasing the average number of items per transaction.

Question 2 (compulsory):
TTR would like to obtain insight in how the unit price varies for the product types that they sell. Conduct
appropriate analysis, report relevant descriptive statistics and use visualizations when reporting your
findings. Include skewness in your interpretation.

In [10]:
(3.0 + 3.75 + 3.15 + 4.5+ 6.3+ 3.5 + 3.25 + 2.0 + 4.25)/9

3.744444444444445

3.74 = mean

In [11]:
numbers = [3.0 , 3.75 , 3.15 ,  4.5 , 6.3 ,  3.5 , 3.25 , 2.0 , 4.25]

In [12]:
median2 = np.median (numbers)
print (median2)

3.5


3.5 = median

In [13]:
data['unit_price'].mean()

3.789817835780315

In [14]:
data['unit_price'].median()

3.5

I tried to get the median and mean in 2 ways just to double check. Out of these results we can say see that the mean is bigger than the median which means that the unit prices are right skewed.

measure of variation - range

In [18]:
import scipy.stats as st

In [19]:
import random

In [20]:
import plotly.express as px

In [21]:
import matplotlib.pyplot as plt


In [22]:
x_values =('unit_price')

In [23]:
y_values = ('product_type')

In [24]:
  box_plot = px.box(data, x='product_type', y='unit_price',
                      title="Box Plot of Unit Prices by Product Type")
if box_plot:
    box_plot.show()




In [25]:
px.histogram(data, x ='product_type' , y  = 'unit_price' ,title = "Histogram of unit prices by product type",nbins=20).show()

In [26]:
import plotly.express as px

Based on the charts, the bulk of the data is concentrated towards the left of the distribution, and the tail extends to the right. This suggests a right-skewed (positively skewed) distribution. In a right-skewed distribution, the mean is typically greater than the median, and there may be a few unusually high values stretching out to the right.

Q3: TTR would like to obtain insight in the average revenue per transaction. Because you are working with
sample data, they would like to also report the interval in which the real (population) average revenue
per transaction is likely to fall. Report and explain your analysis and conclusion.

In [34]:
data ['revenue'] = data ['unit_price'] * data ['transaction_qty']

In [35]:
 (data.head)()

transaction_id transaction_date transaction_time  transaction_qty  \
0               1       06/11/2023         13:12:48              1.0   
1               2       06/07/2023         14:45:20              2.0   
2               3       06/10/2023         14:04:44              1.0   
3               4       06/09/2023         07:47:50              2.0   
4               5       06/10/2023         12:06:49              5.0   

   store_id store_location  unit_price product_category  \
0       2.0        Utrecht        3.00           Bakery   
1       3.0      Eindhoven        3.75           Bakery   
2       3.0      Eindhoven        3.15              Tea   
3       3.0      Eindhoven        4.50        Chocolate   
4       2.0        Utrecht        6.30           Coffee   

             product_type  revenue  
0                   Scone     3.00  
1                Sandwich     7.50  
2               Green tea     3.15  
3           Hot chocolate     9.00  
4  Organic blended coffee    31.50

In [36]:
print("Total amount of transactions:", len(data))

Total amount of transactions: 7356


In [37]:
total_money_spent = (data['unit_price']* data["transaction_qty"]).sum()
print("Total amount of money spent:", total_money_spent )
print("Average revenue per transaction:", (total_money_spent/len(data)))

Total amount of money spent: 54680.100000000006
Average revenue per transaction: 7.433401305057097


In [38]:
unit_price_per_date = data.groupby('transaction_date')["revenue"].sum()

In [39]:
print("The total amount of money spent per day:")
print(unit_price_per_date)


The total amount of money spent per day:
transaction_date
06/05/2023     2780.75
06/06/2023     6211.40
06/07/2023     6353.65
06/08/2023     7727.30
06/09/2023    10423.15
06/10/2023    11596.25
06/11/2023     9587.60
Name: revenue, dtype: float64


In [40]:
amount_of_transactions_per_day = data.groupby(data['transaction_date']).size()
print(amount_of_transactions_per_day)

transaction_date
06/05/2023     379
06/06/2023     799
06/07/2023     871
06/08/2023    1096
06/09/2023    1403
06/10/2023    1537
06/11/2023    1271
dtype: int64


In [41]:
average_value06_05 = ( 2780.75 / 379 )
average_value06_06 = (6211.40 / 799)
average_value06_07 = (  6353.65 / 871 )
average_value06_08 = (7727.30 / 1096 )
average_value06_09 = (  10423.15 / 1403)
average_value06_10 = (11596.25 / 1537 )
average_value06_11 = ( 9587.60 / 1271 )


In [42]:
print ("average value per T 06/05:", average_value06_05)
print ("average value per T 06/06:", average_value06_06)
print ("average value per T 06/07:", average_value06_07)
print ("average value per T 06/08:", average_value06_08)
print ("average value per T 06/09:", average_value06_09)
print ("average value per T 06/10:", average_value06_10)
print ("average value per T 06/11:", average_value06_11)

average value per T 06/05: 7.337071240105541
average value per T 06/06: 7.773967459324155
average value per T 06/07: 7.294661308840413
average value per T 06/08: 7.050456204379562
average value per T 06/09: 7.429187455452602
average value per T 06/10: 7.544729993493819
average value per T 06/11: 7.5433516915814325


In [43]:
sample_mean = ( average_value06_05 + average_value06_06 + average_value06_07 + average_value06_08 + average_value06_09 + average_value06_10 + average_value06_11 ) / 7

In [44]:
print("sample mean:", sample_mean)

sample mean: 7.424775050453932


In [45]:
Average_value_perdays = average_value06_05 , average_value06_06, average_value06_07 , average_value06_08, average_value06_09, average_value06_10 , average_value06_11

In [46]:
std_deviation = np.std ( Average_value_perdays, ddof = 0)
print ("Standard deviation:", std_deviation)

Standard deviation: 0.2121793576010594


In [47]:
Confidence_interval = sample_mean + 1.96 * (std_deviation/7)
print ( Confidence_interval)

7.484185270582229


Assignment 4


How well do different product
types sell in terms of revenue?
Are there product types that sell
significantly worse than others?

In [76]:
data ['total_revenue'] = data ['transaction_qty'] * data ['unit_price']


In [77]:
revenue_by_producttype = data.groupby('product_type')['total_revenue'].sum().sort_values(ascending=False).reset_index()
print(revenue_by_producttype)

              product_type  total_revenue
0            Hot chocolate       16272.00
1   Organic blended coffee        5959.80
2                      Pie        4466.00
3                 Sandwich        4320.00
4             Syrup coffee        3633.75
5                    Scone        3474.00
6               Herbal tea        3300.50
7                 Chai tea        3068.00
8              Milk coffee        2947.00
9                Green tea        2857.05
10               Black tea        2778.00
11            Black coffee        1604.00


In [73]:
mean_revenue_by_producttype = (16272.00 + 5959.80 +4466.00+4320.00+ 3633.75+3474.00+3300.50+3068.00+2947.00+2857.05+2778.00 + 1604.00) / 12
print (mean_revenue_by_producttype)

4556.675
